src/freenet/clients/http/WelcomeToadlet.java

```diff
@@ -265,6 +265,18 @@ public void handleMethodPOST(URI uri, HTTPRequest request, ToadletContext ctx) t
             writeHTMLReply(ctx, 200, "OK", pageNode.generate());
             request.freeParts();
             bucket.free();
+            return;
+        } else if (request.isPartSet("key")) {
+            if(!ctx.checkFormPassword(request)) return;
+            String key;
+            try {
+              key = URLDecoder.decode(new FreenetURI(request.getPartAsStringFailsafe("key", 256)).toURI("/").toString(), false);
```

Comment:
FreenetURI imposes a limit of 32767 (= Short.MAX_VALUE) to the key's length. Any reason to limit it to less than that?